examples/oneview_appliance_device_snmp_v1_trap_destinations.yml

```diff
@@ -13,52 +13,59 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 ###
+# "name" attribute is mandatory in module params and user should provide destination address for name
 ---
 - hosts: all
   vars:
-    - config: "{{ playbook_dir }}/oneview_config.json"
+    config: "{{ playbook_dir }}/oneview_config.json"
+    destination_address: "<ip_address>"
   tasks:
     - name: Create or Update an Appliance Device SNMPv1 Trap Destination by Destination Address
       oneview_appliance_device_snmp_v1_trap_destinations:
         config: "{{ config }}"
         state: present
+        name: "{{ destination_address }}"
         data:
           communityString: "public"
-          destination: "10.0.0.1"
+          destination: "{{ destination_address }}"
           port: 162
       delegate_to: localhost
 
     - debug:
         var: appliance_device_snmp_v1_trap_destinations
 
-    - name: Create or Update an Appliance Device SNMPv1 Trap Destination by URI
+    - name: Ensure Appliance Device SNMPv1 Trap Destination by Destination Address is already present
       oneview_appliance_device_snmp_v1_trap_destinations:
         config: "{{ config }}"
         state: present
+        name: "{{ destination_address }}"
         data:
-          communityString: "private"
-          uri: "/rest/appliance/trap-destinations/1"
+          communityString: "public"
+          destination: "{{ destination_address }}"
           port: 162
       delegate_to: localhost
 
-    - debug:
-        var: appliance_device_snmp_v1_trap_destinations
+    - name: Update Appliance Device SNMPv1 Trap Destination by port
+      oneview_appliance_device_snmp_v1_trap_destinations:
+        config: "{{ config }}"
+        state: present
+        name: "{{ destination_address }}"
+        data:
+          communityString: "public"
+          destination: "{{ destination_address }}"
```

Comment:
Why do we pass same value destination address to 2 different fields name and destination?

Reply:
destination field in data is part of json structure. I have added name field because in "set_resource_object" method, we looks for name and uri fields so that it calls "get_by_name" and "get_by_uri" methods respectively, thereby defining self.current_resource object.